.github/workflows/autoversion.yml

```diff
@@ -0,0 +1,26 @@
+on:
+  schedule:
+    - cron: "30 2 * * *"
+name: Auto-version
+jobs:
+  auto-version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Init Hermit
+        run: |
+          ./bin/hermit env -r >> $GITHUB_ENV
+      - name: Auto-version
+        env:
+          GITHUB_TOKEN: ${{ secrets.AUTO_VERSIONING }}
+        run: |
+          set -xe
+          hermit -d manifest auto-version *.hcl
+          if ! git diff --exit-code; then
+            git config --global user.email "github-actions[bot]@users.noreply.github.com"
+            git config --global user.name "github-actions[bot]"
+            git commit -am   "Auto-versioned: $(git diff --name-only | paste -s -d ' ' -)"
+            git push
```

Comment:
will this work if the default branch (master) is protected? if so, i wonder how?